libcaf_core/caf/detail/parser/read_floating_point.hpp

```diff
@@ -30,15 +30,15 @@ namespace caf::detail::parser {
 ///                    the pre-decimal value.
 template <class State, class Consumer, class ValueType>
 void read_floating_point(State& ps, Consumer&& consumer,
-                         optional<ValueType> start_value,
+                         std::optional<ValueType> start_value,
                          bool negative = false) {
   // Any exponent larger than 511 always overflows.
   static constexpr int max_double_exponent = 511;
   // We assume a simple integer until proven wrong.
   enum sign_t { plus, minus };
   sign_t sign;
   ValueType result;
-  if (start_value == none) {
+  if (start_value == std::nullopt) {
```

Comment:
```suggestion
  if (!start_value) {
```

Nitpick: comparing to `nullopt` does the same thing but is quite verbose.